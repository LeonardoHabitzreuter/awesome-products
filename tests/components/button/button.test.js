import React from 'react'
import { shallow } from 'enzyme'
import Button from '../../../src/components/button'

describe('button component', () => {
  test('should renders properly', () => {
    const wrapper = shallow(<Button type='button' className='btn btn-danger'>Test</Button>)
    expect(wrapper).toMatchSnapshot()
  })

  test('should has submit type and primary class as default props', () => {
    const wrapper = shallow(<Button />)
    expect(wrapper.props()).toMatchObject({
      type: 'submit',
      className: 'btn btn-primary'
    })
  })

  test('should has the same type and style passed through the props', () => {
    const wrapper = shallow(<Button type='button' className='btn btn-danger' />)
    expect(wrapper.props()).toMatchObject({
      type: 'button',
      className: 'btn btn-danger'
    })
  })

  test('should render the children when passed', () => {
    const wrapper = shallow(<Button type='button' className='btn btn-danger'>Test</Button>)
    expect(wrapper.text()).toBe('Test')
  })
})
