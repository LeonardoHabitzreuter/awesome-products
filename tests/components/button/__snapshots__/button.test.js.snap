// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component should renders properly 1`] = `
<button
  className="btn btn-danger"
  type="button"
>
  Test
</button>
`;
